astropy/units/format/cds.py

```diff
@@ -308,41 +310,31 @@ def parse(cls, s, debug=False):
                 else:
                     raise ValueError("Syntax error")
 
+    @classmethod
+    def format_exponential_notation(cls, val, format_spec=".8g"):
+        m, ex = utils.split_mantissa_exponent(val)
+        parts = []
+        if m not in ("", "1"):
+            parts.append(m)
+        if ex:
+            if not ex.startswith("-"):
+                ex = "+" + ex
+            parts.append(f"10{cls._format_superscript(ex)}")
+        return cls._times.join(parts)
```

Comment:
```suggestion
       if ex and not ex.startswith("-"):
            ex = "+" + ex 
        return cls._times.join(
            ((m,) if m in ("", "1") else ())
            + ((f"10{cls._format_superscript(ex)}",) if ex else ())
        )
```

Should be a mild speed bump as it reduces intermediates and extending lists

Reply:
I think this makes it quite a bit less readable, and not obviously faster given the creation and addition of tuples... Since it is not in a fast path anyway, I prefer to just stick with what was there...

Reply:
SGTM.